spec/unit/util/run_mode_spec.rb

```diff
@@ -2,6 +2,13 @@
 require 'spec_helper'
 
 describe Puppet::Util::RunMode do
+
+  # Discriminator for tests that attempts to unset HOME since that, for reasons currently unknown,
```

Comment:
That's because the tests you do doesn't reveal the problem. Try
```
File.expand_path('~/tmp')
```
and you will discover that it expands the path even when you have set the ENV['HOME'] to nil. Prior to 2.4.0, you'd get an exception and that exception is what the disabled tests rely on.